content/blog/2022-07-19-serving-models-with-mlem.md

```diff
@@ -0,0 +1,289 @@
+date: 2022-07-19
+description: >
+  Once you have a machine learning model that's ready for production, getting it
+  out can be complicated. In this tutorial, we're going to use MLEM to deploy a
+  model as a web API.
+descriptionLong: >
+  Getting a machine learning model out to our end users can be an involved task.
+  You have to make sure it's in the correct format for the environment and for
+  the way it will be consumed. We're going to walk through how you can use MLEM
+  to save your model and serve it with a web API.
+picture: 2022-07-19/serve_mlem_models.png
+pictureComment: Serving Machine Models with MLEM
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/serving-machine-learning-models-with-mlem/1253
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - MLEM
+  - Collaboration
+---
+
+Training a machine learning model is only one step in the process of getting
+something useful out to end-users. When it's time to deploy the model to
+production, there are a number of approaches you can take depending on the goal
+of the machine learning project. That might mean getting the model ready to
+respond to real-time queries coming from an API or batch processing predictions,
+for example.
+
+Either way, you'll need to save your trained and validated model in a format
+that's consumable by other systems. That's why we'll be covering how to serve
+models through a [REST](https://restfulapi.net/) endpoint or a Python package
+with [MLEM](https://mlem.ai/).
+
+> You can get the repo we're working with
+> [here](https://github.com/iterative/stale-model-example/tree/mlem-serve).
+
+## Take a candidate model
+
+There are instructions in the project
+[README](https://github.com/iterative/stale-model-example/tree/mlem-serve#readme)
+on how to get everything you need installed and running. This is a simple ML
+project that uses [DVC](https://dvc.org/) for data versioning and experiment
+tracking.
+
+After you have the repo set up, you'll already have the `MLEM` package
+installed. This project already has a model that's been trained and validated so
+we can move on to saving this model.
+
+## Save the model
+
+Inside the `train.py` script, we need to add the `MLEM` import to save the
+models as we experiment. We don't have to worry about running the training
+script for this project since we have the model, but it's good to know what's
+happening under the hood.
+
+```python
+# train.py
+import os
+import pickle5 as pickle
+import sys
+import yaml
+
+from mlem.api import save
+import numpy as np
+from sklearn.ensemble import RandomForestClassifier
+...
+```
+
+Then you can add the `save` function to the end of the training script.
+
+```python
+# train.py
+...
+clf = RandomForestClassifier(
+    n_estimators=n_est, min_samples_split=min_split, n_jobs=2, random_state=seed
+)
+
+clf.fit(x, labels)
+
+save(
+    clf,
+    "clf",
+    sample_data=x,
+    description="Random Forest Classifier",
+)
+```
+
+<admon type="tip">
```

Comment:
is it intentional the tip takes a few screens? I would advice against this. Use an expandable section instead. Admons should be small in size (they should should jump out of the text, not be text itself)